src/qd8-f32-qb4w-gemm/gen/qd8-f32-qb4w-gemm-1x4-minmax-scalar.c

```diff
@@ -0,0 +1,169 @@
+// Auto-generated file. Do not edit!
+//   Template: src/qs8-gemm/scalar.c.in
+//   Generator: tools/xngen
+//
+// Copyright 2021 Google LLC
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include <assert.h>
+
+#include <xnnpack/gemm.h>
+#include <xnnpack/math.h>
+
+
+void xnn_qd8_f32_qb4w_gemm_minmax_ukernel_1x4__scalar(
+    size_t mr,
+    size_t nc,
+    size_t kc,
+    const int8_t* restrict a,
+    size_t a_stride,
+    const void* restrict w,
+    float* restrict c,
+    size_t cm_stride,
+    size_t cn_stride,
+    const union xnn_f32_qb4w_minmax_params params[restrict XNN_MIN_ELEMENTS(1)],
+    const struct xnn_qd8_quantization_params quantization_params[restrict XNN_MIN_ELEMENTS(1)])
+{
+  assert(mr != 0);
+  assert(mr <= 1);
+  assert(nc != 0);
+  assert(kc != 0);
+  size_t bl = params->scalar.blocksize;
+  assert(bl <= round_up_po2(kc, 2));
+  assert(bl != 0);
+
+  const int8_t* a0 = a;
+  float* c0 = c;
+
+  kc = round_up_po2(kc, 2);
+  do {
+    const float vksum0 = ((const float*) w)[0];
+    const float vksum1 = ((const float*) w)[1];
+    const float vksum2 = ((const float*) w)[2];
+    const float vksum3 = ((const float*) w)[3];
+    const float vinput_zero_point0 = (const float) quantization_params[0].zero_point;
+    float vout0x0 = vksum0 * vinput_zero_point0;
+    float vout0x1 = vksum1 * vinput_zero_point0;
+    float vout0x2 = vksum2 * vinput_zero_point0;
+    float vout0x3 = vksum3 * vinput_zero_point0;
+    w = (const float*) w + 4;
+
+    size_t n_blocks = kc / bl;
+    for (size_t nb=0; nb<n_blocks; ++nb) {
+      int32_t vacc0x0 = 0;
+      int32_t vacc0x1 = 0;
+      int32_t vacc0x2 = 0;
+      int32_t vacc0x3 = 0;
+      size_t k = bl;
+    for (; k >= 2 * sizeof(uint8_t); k -= 2 * sizeof(uint8_t)) {
```

Comment:
Can you please fix indentation?

Reply:
I suspect this may not be trivial given the loop or the indent itself is optional. But I will let Gregory try it out.

Reply:
@fbarchard do you know any tricks for this?

Reply:
I was thinking another option might be to introduce a keyword similar to `LINT` markers in `xngen.py` to manipulate indent from the `.in` files.